src/masternodes/rpc_loan.cpp

```diff
@@ -0,0 +1,251 @@
+#include <masternodes/mn_rpc.h>
+
+UniValue setCollateralTokenToJSON(CLoanSetCollateralTokenImplementation const& collToken)
+{
+    UniValue collTokenObj(UniValue::VOBJ);
+
+    auto token = pcustomcsview->GetToken(collToken.idToken);
+        if (!token)
+            return (UniValue::VNULL);
+    collTokenObj.pushKV("token", token->CreateSymbolKey(collToken.idToken));
+    collTokenObj.pushKV("factor", ValueFromAmount(collToken.factor));
+    collTokenObj.pushKV("priceFeedId", collToken.priceFeedTxid.GetHex());
+    if (collToken.activateAfterBlock)
+        collTokenObj.pushKV("activateAfterBlock", static_cast<int>(collToken.activateAfterBlock));
+
+    UniValue ret(UniValue::VOBJ);
+    ret.pushKV(collToken.creationTx.GetHex(), collTokenObj);
+    return (ret);
+}
+
+UniValue setcollateraltoken(const JSONRPCRequest& request) {
+    CWallet* const pwallet = GetWallet(request);
+
+    RPCHelpMan{"setcollateraltoken",
+                "Creates (and submits to local node and network) a set colleteral token transaction.\n" +
+                HelpRequiringPassphrase(pwallet) + "\n",
+                {
+                    {"parameters", RPCArg::Type::OBJ, RPCArg::Optional::NO, "",
+                        {
+                            {"token", RPCArg::Type::STR, RPCArg::Optional::NO, "Symbol or id of collateral token"},
+                            {"factor", RPCArg::Type::NUM, RPCArg::Optional::NO, "Collateralization factor"},
+                            {"priceFeedId", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "txid of oracle feeding the price"},
+                            {"activateAfterBlock", RPCArg::Type::NUM, RPCArg::Optional::OMITTED, "changes will be active after the block height (Optional)"},
+                        },
+                    },
+                    {"inputs", RPCArg::Type::ARR, RPCArg::Optional::OMITTED_NAMED_ARG,
+                        "A json array of json objects",
+                        {
+                            {"", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED, "",
+                                {
+                                    {"txid", RPCArg::Type::STR_HEX, RPCArg::Optional::NO, "The transaction id"},
+                                    {"vout", RPCArg::Type::NUM, RPCArg::Optional::NO, "The output number"},
+                                },
+                            },
+                        },
+                    },
+                },
+                RPCResult{
+                        "\"hash\"                  (string) The hex-encoded hash of broadcasted transaction\n"
+                },
+                RPCExamples{
+                        HelpExampleCli("setcollateraltoken", "'{\"ownerAddress\":\"<tokenAddress>\","
+                                                        "\"tokenFrom\":\"GOLD#128\",\"chainTo\":\"BTC\","
+                                                        "\"amountFrom\":\"10\",\"orderPrice\":\"10\"}'")                },
```

Comment:
The example command is incorrect

Reply:
Should be resolved here.

https://github.com/DeFiCh/ain/pull/581/commits/731a77fbc1f9afc9345ae290cde9b9cc66ee45fb